### PR TITLE
NXBT-1770: T&P Add Dockerfile and docker-compose

### PR DESCRIPTION
**Pull request by alexistimic**
Add Dockerfile-swarm to build nuxeo/jenkins-swarm in order to have an image that can connect to the jenkins swarm

### CODE REVIEW

File: docker/Dockerfile-swarm [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM dockerpriv.nuxeo.com:443/nuxeo/jenkins-slave
+RUN wget https://maven-eu.nuxeo.org/nexus/service/local/repositories/vendor-releases/content/org/jenkins-ci/plugins/swarm-client/3.12/swarm-client-3.12-remoting-2-59-2.jar -O /opt/jenkins/bin/swarm-client.jar
```
Comment (jcarsique) [inline docker/Dockerfile-swarm]: maven EU vs US vs maven.nuxeo.org ?

File: docker/Dockerfile-swarm [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+COPY $PWD/docker/swarm.sh /opt/jenkins/bin/swarm-client.sh
```
Comment (jcarsique) [inline docker/Dockerfile-swarm]: why PWD?

File: docker/docker-compose-swarm.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  swarm:
+    build:
+       context: ${SWARM}/jenkins-slave-swarm
+       dockerfile: Dockerfile
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock:rw
+      - workspace:/opt/jenkins/workspace
+    environment:
+      - JAVA_OPTS=-Xms512m -Xmx3g
+    command: /sbin/my_init -- su - jenkins -c "java -jar /opt/jenkins/bin/swarm-client.jar -master $JENKINS_MASTER -name $COMPOSE_PROJECT_NAME -deleteExistingClients  -disableClientsUniqueId -retry 3 -username jenkins -password $JENKINS_API_TOKEN -labels swarm"
```
Comment (jcarsique) [inline docker/docker-compose-swarm.yml]: parametrize the username too

File: docker/swarm.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh -x
+
+env
+
+su - jenkins -c "java -jar \"/opt/jenkins/bin/swarm-client.jar\" -master $JENKINS_MASTER -name $NAME -deleteExistingClients  -disableClientsUniqueId -retry 3 -username jenkins -password $JENKINS_API_TOKEN -labels swarm"
```
Comment (jcarsique) [inline docker/swarm.sh]: parametrize the username too

File: docker/docker-compose-swarm.yml (hunk 3 above)
Comment (alexistimic) [reply docker/docker-compose-swarm.yml]: Done

File: docker/swarm.sh (hunk 4 above)
Comment (alexistimic) [reply docker/swarm.sh]: Done

**Comment by alexistimic** [comment]
- create a specific ticket holding that new feature => NXBT-2336
- add headers to files => Done
- add usage or complete README => Add README

File: docker/Dockerfile-swarm (hunk 2 above)
Comment (alexistimic) [reply docker/Dockerfile-swarm]: Replaced PWD by relative Path

File: docker/Dockerfile-swarm (hunk 1 above)
Comment (alexistimic) [reply docker/Dockerfile-swarm]: Replaced maven-eu by maven.nuxeo.org

File: docker/swarm/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Contributors:
+#     slacoin
+#     atimic
+#
+
+
+FROM dockerpriv.nuxeo.com:443/nuxeo/jenkins-slave
```
Comment (jcarsique) [inline docker/swarm/Dockerfile]: Unless I'm wrong this is the first change making the script unusable outside.
The only other references were in the README.

File: docker/swarm/README.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# Swarm 
+
+Docker image based on nuxeo jenkins public slave that have jenkins-swarm-client integrated
```
Comment (jcarsique) [inline docker/swarm/README.md]: `nuxeo jenkins public slave` ?
jenkins-swarm-client deserves links and pointers...

**Pull request merged**